src/main/java/com/o19s/es/ltr/feature/store/ScriptFeature.java

```diff
@@ -233,7 +233,7 @@ public Query doToQuery(LtrQueryContext context, FeatureSet featureSet, Map<Strin
         Script script = new Script(this.script.getType(), this.script.getLang(),
             this.script.getIdOrCode(), this.script.getOptions(), nparams);
         ScoreScript.Factory factoryFactory  = context.getQueryShardContext().compile(script, ScoreScript.CONTEXT);
-        ScoreScript.LeafFactory leafFactory = factoryFactory.newFactory(nparams, context.getQueryShardContext().lookup());
+        ScoreScript.LeafFactory leafFactory = factoryFactory.newFactory(nparams, context.getQueryShardContext().lookup(), context.getQueryShardContext().searcher());
```

Comment:
this is more about the newFactory function, than this review; does it make sense that we're now asking for two parameters from the same context.getQueryShardContext()? Could this parameter list continue to grow?

Reply:
> this is more about the newFactory function, than this review; does it make sense that we're now asking for two parameters from the same context.getQueryShardContext()?

The factory does not know about `QueryShardContext`, it only needs lookup and searcher that could be provided by the context.